swri_geometry_util/CMakeLists.txt

```diff
@@ -32,7 +32,7 @@ target_include_directories(${PROJECT_NAME}
   ${GEOS_INCLUDE_DIRS}
 )
 target_link_libraries(${PROJECT_NAME}
-  ${GEOS_LIBRARIES}
+  geos
```

Comment:
Unfortunately, the libgeos C++ package is broken in Ubuntu 22.04.  This will compile if you have tests disabled, but if you enable tests, or if anything else tries to use this library, it will fail to link like so:
```
/usr/bin/ld: libswri_geometry_util.so: undefined reference to `geos::geom::Coordinate::Coordinate(double, double, double)'
/usr/bin/ld: libswri_geometry_util.so: undefined reference to `geos::geom::Coordinate::Coordinate(double, double, double)'
collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/test_intersection.dir/build.make:183: test_intersection] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:187: CMakeFiles/test_intersection.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake[2]: *** [CMakeFiles/test_geometry_util.dir/build.make:183: test_geometry_util] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:215: CMakeFiles/test_geometry_util.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

Looking through the symbols in the library with `nm -D -C /usr/lib/x86_64-linux-gnu/libgeos.so.3.10.2` reveals that there are no constructors for the `Coordinate` class, so it cannot actually be used.

See https://bugs.launchpad.net/ubuntu/+source/geos/+bug/1980147 for some more details.  The upstream GEOS developers have decided that the package is intentionally broken and they're not going to fix it.

Pretty much the only solution for this is to remove the functions that use libgeos or rewrite them to use the C API.

Reply:
Agreed. If we do this, I think we just make it use the C API for all distros.